interface/expand_contract_js.php

```diff
@@ -0,0 +1,51 @@
+<?php
+  /**
+ * expand contract jquery script
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author Ranganath Pathak <pathak@scrs1.org>
+ * @copyright Copyright (c) 2018 Ranganath Pathak <pathak@scrs1.org>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+// ensure that $user_settings_php_path, $arr_files_php variables are set in the script calling this script
+?>
+$( document ).ready(function() {
+    $('.expand_contract').click(function() {
+        var elementTitle = $(this).prop('title');
+        var contractTitle = '<?php echo xla('Click to Contract and set to henceforth open in Centered mode'); ?>';
+        var expandTitle = '<?php echo xla('Click to Expand and set to henceforth open in Expanded mode'); ?>';
+        //var userSettingsPath = '<?php echo $GLOBALS['webroot'] . "/library/ajax/user_settings.php"; ?>';
+        var arrFiles = [];
+        
+        <?php
+        if (strpos($arr_files_string, ",")) {?>
+              arrFiles = '<?php echo attr($arr_files_string);?>'.split(",");
+            <?php
+        } else {?>
+             arrFiles = ['<?php echo attr($arr_files_string);?>'];
+            <?php
+        } ?>
```

Comment:
Could consider something like (which directly uses the php array):
```
arrFiles = [<?php
$first_file_php = true;
foreach ($arr_files_php as $file_php) {
    if ($first_file_php) {
        echo "'" . attr($file_php) . "'";
        $first_file_php = false;
    } else {
        echo ",'" . attr($file_php) . "'";
    }
}
?>]
```

Reply:
@bradymiller 

1. The problem with escaping the variables as I see it is in this bit of code
 `$.post( '<?php echo $user_settings_php_path; ?>', { target: arrFiles[index].trim(), setting: 0 });`
When these 2 variables, `arrFiles[index].trim()`, `setting: 0` , are posted they cannot be escaped using `attr`

Instead I have applied `attr` to the POSTed values in `library/ajax/user_settings.php` like this
 `setUserSetting(attr($_POST['target']), attr($_POST['setting']));`

2. I solved the problem of converting the php array $arr_files_php into the  JS array arrFiles by replacing 

```
// needed for the jQuery part of script
    if (count($arr_files_php) > 1) {
        $arr_files_string = implode ("," , $arr_files_php);
    } else {
        $arr_files_string = $arr_files_php[0];
    }
```
in each individual file 

and  
```
var arrFiles = [];
 <?php
if(strpos($arr_files_string, ",")) {?>
    arrFiles = '<?php echo attr($arr_files_string);?>'.split(",");
<?php
} else {?>
    arrFiles = ['<?php echo attr($arr_files_string);?>'];
<?php
} ?>

```
in `interface/expand_contract_js.php`

with a single line of code 
`var arrFiles = <?php echo json_encode($arr_files_php) ?>;`
in `interface/expand_contract_js.php`

Hope this passes muster :)